pkg/operator/workloadcontroller/workload_controller.go

```diff
@@ -144,6 +147,21 @@ func (c OpenShiftAPIServerOperator) sync() error {
 		return nil
 	}
 
+	// block until extension-apiserver-authentication configmap is available
+	// see https://bugzilla.redhat.com/show_bug.cgi?id=1795163#c19 to check why we have to wait for it
+	// in the future we need to change upstream code to be more dynamic
+	if !c.haveObservedExtensionConfigMap {
+		_, err = c.kubeClient.CoreV1().ConfigMaps(metav1.NamespaceSystem).Get("extension-apiserver-authentication", metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			klog.Infof("Waiting for %q configmap in %q namespace to be available", "extension-apiserver-authentication", metav1.NamespaceSystem)
```

Comment:
this delays the deployment by up to a minute, doesn't it?

Reply:
> this delays the deployment by up to a minute, doesn't it?

the kube-apiserver (even the bootstrap one), should create this configmap. it should never practically delay.  If it does, then we want to know.

Reply:
Our sync is blocked for that time. That's what I mean. Would be good to be clear how this gets requeued.